docker/runit/init.sh

```diff
@@ -32,5 +32,19 @@ sed -i "s,<PROTOCOL>,${SPARK_SSL_PROTOCOL}," /etc/nginx/conf.d/spark.conf
 # bytes cipher strings of Java.
 # sed -i "s,<ENABLED_ALGORITHMS>,${SPARK_SSL_ENABLEDALGORITHMS//,/:}," /etc/nginx/conf.d/spark.conf
 
-# start service
-exec runsvdir -P /etc/service
+# Start runit processes (spark and nginx).
+exec runsvdir -P /etc/service &
+
+# The Spark process stdout is logged by runit's svlogd to a file named
+# "current".
+readonly SPARK_STDOUT_FILE="${MESOS_SANDBOX}"/spark/current
+
+# Wait for Spark process log file to exist.
+while [ ! -f "${SPARK_STDOUT_FILE}" ]; do
+  sleep 1
```

Comment:
won't this cause an infinite loop in case dispatcher fails to start and doesn't create a file?